src/test/java/bio/terra/workspace/integration/WorkspaceIntegrationTest.java

```diff
@@ -28,24 +29,91 @@
 
+  @Autowired private WorkspaceManagerTestClient workspaceManagerTestClient;
   @Autowired private TestUtils testUtils;
-
   @Autowired private TestConfiguration testConfig;
-
   @Autowired private AuthService authService;
+  private final String TAG_NEEDS_CLEANUP = "needs-cleanup";
+  private static final Logger logger = LoggerFactory.getLogger(WorkspaceIntegrationTest.class);
+
+  // Variable for workspace Id's to clean up in tearDown(TestInfo testInfo) method, after
+  // running applicable test method(s)
+  private UUID workspaceIdToCleanup;
 
   @BeforeEach
   public void setup() {}
 
+  /*
+   * TODO: Cloud Recourse Library (CRL) is planning to broadly address resource cleanup for integration tests in MC Terra
+   *  applications.
+   *  The doc below proposes a "Resource Tracking" approach for cleaning up. We should follow the developments in CRL for
+   *  potential utilization here in Workspace Manager.
+   *  Doc: https://docs.google.com/document/d/13mYVJML_fOLsX1gxQxRJgECUNT27dAMKzEJxUEvtQqM/edit#heading=h.x06ofvfgp7wt
+   */
+  @AfterEach
+  public void tearDown(TestInfo testInfo) throws Exception {
+    Set<String> tags = testInfo.getTags();
+    if (tags != null && tags.contains(TAG_NEEDS_CLEANUP)) {
+      cleanUpWorkspace();
+    }
+  }
+
   @Test
+  @Tag(TAG_NEEDS_CLEANUP)
   public void createWorkspace() throws Exception {
-    // todo: Seeing that we have https://broadworkbench.atlassian.net/browse/AS-315 in the upcoming,
-    //  Let's reuse the code to-be-written there to clean up workspaces after creating them
-    //  Clean up manually for now
     UUID workspaceId = UUID.randomUUID();
-    String path = testConfig.getWsmCreateWorkspaceUrl();
+    workspaceIdToCleanup = workspaceId;
+    WorkspaceResponse<CreatedWorkspace> workspaceResponse = createDefaultWorkspace(workspaceId);
+
+    Assertions.assertEquals(workspaceResponse.getStatusCode(), HttpStatus.OK);
+    Assertions.assertTrue(workspaceResponse.isResponseObject());
+    CreatedWorkspace createdWorkspace = workspaceResponse.getResponseObject();
+    Assertions.assertEquals(workspaceId.toString(), createdWorkspace.getId());
+  }
+
+  @Test
+  public void deleteWorkspace() throws Exception {
+    UUID workspaceId = UUID.randomUUID();
+    WorkspaceResponse<CreatedWorkspace> workspaceResponse = createDefaultWorkspace(workspaceId);
+
+    String userEmail = testConfig.getServiceAccountEmail();
+    String token = authService.getAuthToken(userEmail);
+    String path = testConfig.getWsmWorkspacesBaseUrl() + "/" + workspaceId;
+    DeleteWorkspaceRequestBody body = new DeleteWorkspaceRequestBody().authToken(token);
+    String jsonBody = testUtils.mapToJson(body);
+
+    WorkspaceResponse<?> deleteWorkspaceResponse =
+        workspaceManagerTestClient.delete(userEmail, path, jsonBody);
+
+    Assertions.assertEquals(deleteWorkspaceResponse.getStatusCode(), HttpStatus.NO_CONTENT);
+  }
+
+  @Test
+  @Tag(TAG_NEEDS_CLEANUP)
```

Comment:
if you wanted to, you could create a custom `@NeedsCleanup` annotation, which could be reusable across many test classes: https://junit.org/junit5/docs/current/user-guide/#writing-tests-meta-annotations

Reply:
Created AS-431 for this, as it looks like we may need to annotate a few other things.